src/mudlet.cpp

```diff
@@ -2879,13 +3039,80 @@ void mudlet::doAutoLogin(const QString& profile_name)
         mudlet::self()->setupPreInstallPackages(pHost->getUrl().toLower());
     }
 
+    // May be needed
+    pHost->mTelnet.setEncoding(readProfileData(profile_name, QStringLiteral("encoding")).toUtf8(), false);
+
     emit signal_hostCreated(pHost, mHostManager.getHostCount());
     emit signal_adjustAccessibleNames();
     slot_connectionDialogueFinished(profile_name, true);
     enableToolbarButtons();
     updateMultiViewControls();
 }
+void mudlet::handleTelnetUri(const QUrl& telnetUri)
+{
+    QUrl url(telnetUri);
 
+    if (url.port() == -1) {
+        url.setPort(23);
+    }
+
+    QStringList hostList = QDir(getMudletPath(profilesPath)).entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+
+    int profilesFound{};
```

Comment:
:vomiting_face: - for complex classes I suppose the default/empty initialisation with `{}` does look attractive but for plain old data types like this I honestly think this is better:
```suggestion
    int profilesFound = 0;
```